From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harmer <mark.harmer@discordap.com>
Date: Sun, 21 Jan 2024 17:12:34 -0800
Subject: Revert "[task] Stop using NSRunLoop in renderer process"

This reverts commit 1fb1e52dd81114e46ab5db5362492f2f35db7d6b.

discord_utils keyboard input requires the macos CFRunLoop message pump to still be running in the renderer process.

diff --git a/content/renderer/renderer_main.cc b/content/renderer/renderer_main.cc
index 47d0eed6bc95cb7b034fe07473defb5a70e95de3..4c9b8c6b0969779b8437d7a8f0ec5ade0eaf56e7 100644
--- a/content/renderer/renderer_main.cc
+++ b/content/renderer/renderer_main.cc
@@ -107,7 +107,12 @@ void HandleRendererErrorTestParameters(const base::CommandLine& command_line) {
 }
 
 std::unique_ptr<base::MessagePump> CreateMainThreadMessagePump() {
-#if BUILDFLAG(IS_FUCHSIA)
+#if BUILDFLAG(IS_MAC)
+  // As long as scrollbars on Mac are painted with Cocoa, the message pump
+  // needs to be backed by a Foundation-level loop to process NSTimers. See
+  // http://crbug.com/306348#c24 for details.
+  return base::MessagePump::Create(base::MessagePumpType::NS_RUNLOOP);
+#elif BUILDFLAG(IS_FUCHSIA)
   // Allow FIDL APIs on renderer main thread.
   return base::MessagePump::Create(base::MessagePumpType::IO);
 #else
